File: frontend/src/app/shared/model/ingredient.ts
export interface IngredientSearchResDto {
  id: number;
  name: string;
  price: number;
  calories: number;
  notes: string;
}

export interface IngredientDetailDto {
  id: number;
  name: string;
  price: number;
  calories: number;
  notes: string;
}

export interface IngredientCreateReqDto {
  name: string;
  price: number;
  calories: number;
  notes: string;
}

export interface IngredientCreateResDto {
  id: number;
  name: string;
  price: number;
  calories: number;
  notes: string;
}


export interface IngredientUpdateReqDto {
  name: string;
  price: number;
  calories: number;
  notes: string;
}

File: backend/src/main/java/com/bartsch/costlydiet/model/entity/Ingredient.java
package com.bartsch.costlydiet.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
public class Ingredient {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column
  private String name;

  @Column
  private long price; // in cents per kg

  @Column
  private long calories;

  @Column
  private String notes;

  @Column
  @CreationTimestamp
  private LocalDateTime createdAt;
}
